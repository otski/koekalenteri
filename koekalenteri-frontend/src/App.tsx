/* eslint-disable mobx/missing-observer */
import { Auth } from '@aws-amplify/auth';
import { Authenticator } from '@aws-amplify/ui-react';
import '@aws-amplify/ui-react/styles.css';
import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { createTheme } from '@mui/material/styles';
import { makeStyles, ThemeProvider } from '@mui/styles';
import { SnackbarProvider } from 'notistack';
import { useTranslation } from 'react-i18next';
import { Navigate, Route, Routes } from 'react-router-dom';
import { AWSConfig } from './amplify-env';
import { ADMIN_DEFAULT, ADMIN_EDIT_EVENT, ADMIN_EVENTS, ADMIN_EVENT_TYPES, ADMIN_JUDGES, ADMIN_NEW_EVENT, ADMIN_OFFICIALS, ADMIN_ORGS, ADMIN_ROOT, ADMIN_USERS, ADMIN_VIEW_EVENT } from './config';
import { Language, locales, muiLocales } from './i18n';
import { EventEditPage, EventListPage, EventRegistrationPage, EventTypeListPage, EventViewPage, JudgeListPage, LoginPage, LogoutPage, OfficialListPage, OrganizerListPage, SearchPage, UsersPage } from './pages';

Auth.configure(AWSConfig);

const useStyles = makeStyles({
  snack: {
    paddingTop: 38 // So snacks appear below the header
  }
});

function App() {
  const { i18n } = useTranslation();
  const language = i18n.language as Language;
  const classes = useStyles();

  return (
    <ThemeProvider theme={(outerTheme) => createTheme(outerTheme, muiLocales[language])}>
      <LocalizationProvider dateAdapter={AdapterDateFns} adapterLocale={locales[language]}>
        <SnackbarProvider
          anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
          classes={{ containerRoot: classes.snack }}
          maxSnack={3}
        >
          <Authenticator.Provider>
            <Routes>
              <Route path="/" element={<SearchPage />} />
              <Route path="/event/:eventType/:id"  element={<EventRegistrationPage />} />
              <Route path="/event/:eventType/:id/:class"  element={<EventRegistrationPage />} />
              <Route path="/event/:eventType/:id/:class/:date" element={<EventRegistrationPage />} />
              <Route path="/login" element={<LoginPage />} />
              <Route path="/logout" element={<LogoutPage />} />
              <Route path={ADMIN_ROOT} element={<Navigate replace to={ADMIN_DEFAULT} />} />
              <Route path={ADMIN_EVENTS} element={<EventListPage />} />
              <Route path={ADMIN_NEW_EVENT} element={<EventEditPage create />} />
              <Route path={`${ADMIN_EDIT_EVENT}/:id`} element={<EventEditPage />} />
              <Route path={`${ADMIN_VIEW_EVENT}/:id`} element={<EventViewPage />} />
              <Route path={`${ADMIN_VIEW_EVENT}/:id/:reistrationId`} element={<EventViewPage />} />
              <Route path={ADMIN_ORGS} element={<OrganizerListPage />} />
              <Route path={ADMIN_OFFICIALS} element={<OfficialListPage />} />
              <Route path={ADMIN_USERS} element={<UsersPage />} />
              <Route path={ADMIN_JUDGES} element={<JudgeListPage />} />
              <Route path={ADMIN_EVENT_TYPES} element={<EventTypeListPage />} />
            </Routes>
          </Authenticator.Provider>
        </SnackbarProvider>
      </LocalizationProvider>
    </ThemeProvider>
  );
}

export default App;
