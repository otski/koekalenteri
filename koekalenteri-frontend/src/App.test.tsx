import { fireEvent, render, screen, within } from '@testing-library/react';
import App from './App';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from '@mui/material';
import theme from './assets/Theme';
import { ADMIN_EVENT_TYPES, ADMIN_JUDGES, ADMIN_NEW_EVENT, ADMIN_ORGS, ADMIN_ROOT, ADMIN_USERS, ADMIN_VIEW_EVENT } from './config';
import { DataGridProps } from '@mui/x-data-grid';

const TIMEOUT = 5000;

jest.mock('./api/event');
jest.mock('./api/judge');
jest.mock('./api/official');
jest.mock('./api/organizer');
jest.mock('./api/eventType');

jest.mock('@aws-amplify/ui-react');

// DataGrid needs disableVirtualizaton to render properly in tests
jest.mock('@mui/x-data-grid', () => {
  const { DataGrid } = jest.requireActual('@mui/x-data-grid');
  return {
    ...jest.requireActual('@mui/x-data-grid'),
    DataGrid: (props: DataGridProps) => {
      return (
        <DataGrid
          {...props}
          disableVirtualization
        />
      );
    },
  };
});

function renderPath(path: string) {
  render(
    <MemoryRouter initialEntries={[path]}>
      <ThemeProvider theme={theme}>
        <App />
      </ThemeProvider>
    </MemoryRouter>
  );
}

it('renders home page', () => {
  renderPath('/');
  const imgElement = screen.getByAltText('Suomen noutajakoirajärjestö');
  expect(imgElement).toBeInTheDocument();
}, TIMEOUT);

it('renders event registration page', async () => {
  renderPath('/event/NOME-B/test2');
  const spinner = screen.getByRole('progressbar');
  expect(spinner).toBeInTheDocument();
  const organizer = await screen.findByText(/Järjestäjä 2/);
  expect(organizer).toBeInTheDocument();
  expect(spinner).not.toBeInTheDocument();
}, TIMEOUT);

it('renders event registration page with class and date selected', async () => {
  renderPath('/event/NOME-B/test2/AVO/13.02.');
  const c = await screen.findByText(/AVO, la \(aamu\) \/ la \(ilta\)/);
  expect(c).toBeInTheDocument();
}, TIMEOUT);

fit('renders admin default (event) page', async () => {
  renderPath(ADMIN_ROOT);
  const spinner = screen.getByRole('progressbar');
  expect(spinner).toBeInTheDocument();
  await screen.findByText('type1');
  expect(spinner).not.toBeInTheDocument();
  const row = screen.getAllByRole('row')[1];
  expect(row).toMatchSnapshot();
  const cell = within(row).getAllByRole('cell')[0];
  fireEvent.click(cell, 'click');
  expect(row).toHaveClass('Mui-selected');
  fireEvent.click(screen.getByText(/Muokkaa/));
  const newHead = await screen.findByText(/Muokkaa tapahtumaa/);
  expect(newHead).toBeInstanceOf(HTMLDivElement);
}, TIMEOUT);

it('renders admin createEvent page', async () => {
  renderPath(ADMIN_NEW_EVENT);
  const head = await screen.findByText(/Uusi tapahtuma/);
  expect(head).toBeInTheDocument();
}, TIMEOUT);

it('renders admin EventViewPage', async () => {
  renderPath(`${ADMIN_VIEW_EVENT}/test1`);
  const head = await screen.findByText(/type1, 10.-11.2.2021, test location/i, undefined, { timeout: TIMEOUT });
  expect(head).toBeInTheDocument();
}, TIMEOUT);

it('renders admin organizations', async () => {
  renderPath(ADMIN_ORGS);
  const head = await screen.findAllByText(/Yhdistykset/);
  expect(head.length).toBe(3);
}, TIMEOUT);

it('renders admin users', async () => {
  renderPath(ADMIN_USERS);
  const head = await screen.findAllByText(/Käyttäjät/);
  expect(head.length).toBe(2);
}, TIMEOUT);

it('renders admin judges', async () => {
  renderPath(ADMIN_JUDGES);
  const head = await screen.findAllByText(/Tuomarit/);
  expect(head.length).toBe(3);
}, TIMEOUT);

it('renders admin eventTypes', async () => {
  renderPath(ADMIN_EVENT_TYPES);
  const head = await screen.findAllByText(/Koemuodot/);
  expect(head.length).toBe(3);
}, TIMEOUT);

it('renders logout page', async () => {
  renderPath('/logout');
  expect(await screen.findByText('Kirjaudu')).toBeInTheDocument();
}, TIMEOUT);
