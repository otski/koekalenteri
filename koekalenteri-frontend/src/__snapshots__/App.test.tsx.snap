// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders admin default (event) page 1`] = `
<div
  aria-rowindex="2"
  aria-selected="false"
  class="MuiDataGrid-row"
  data-id="test1"
  data-rowindex="0"
  role="row"
  style="max-height: 36px; min-height: 36px;"
>
  <div
    aria-colindex="1"
    aria-colspan="1"
    class="MuiDataGrid-cell MuiDataGrid-cell--textRight"
    data-colindex="0"
    data-field="date"
    role="cell"
    style="min-width: 120px; max-width: 120px; min-height: 36px; max-height: 36px;"
    tabindex="-1"
  >
    <div
      class="MuiDataGrid-cellContent"
    >
      10.-11.2.2021
    </div>
  </div>
  <div
    aria-colindex="2"
    aria-colspan="1"
    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
    data-colindex="1"
    data-field="eventType"
    role="cell"
    style="min-width: 100px; max-width: 100px; min-height: 36px; max-height: 36px;"
    tabindex="-1"
  >
    <div
      class="MuiDataGrid-cellContent"
    >
      type1
    </div>
  </div>
  <div
    aria-colindex="3"
    aria-colspan="1"
    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
    data-colindex="2"
    data-field="classes"
    role="cell"
    style="min-width: 0; max-width: 0; min-height: 36px; max-height: 36px;"
    tabindex="-1"
  >
    <div
      class="MuiDataGrid-cellContent"
    >
      class1
    </div>
  </div>
  <div
    aria-colindex="4"
    aria-colspan="1"
    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
    data-colindex="3"
    data-field="location"
    role="cell"
    style="min-width: 0; max-width: 0; min-height: 36px; max-height: 36px;"
    tabindex="-1"
  >
    <div
      class="MuiDataGrid-cellContent"
    >
      test location
    </div>
  </div>
  <div
    aria-colindex="5"
    aria-colspan="1"
    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
    data-colindex="4"
    data-field="official"
    role="cell"
    style="min-width: 0; max-width: 0; min-height: 36px; max-height: 36px;"
    tabindex="-1"
  >
    <div
      class="MuiDataGrid-cellContent"
    />
  </div>
  <div
    aria-colindex="6"
    aria-colspan="1"
    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
    data-colindex="5"
    data-field="judges"
    role="cell"
    style="min-width: 0; max-width: 0; min-height: 36px; max-height: 36px;"
    tabindex="-1"
  >
    <div
      class="MuiDataGrid-cellContent"
    >
      Tuomari 1
    </div>
  </div>
  <div
    aria-colindex="7"
    aria-colspan="1"
    class="MuiDataGrid-cell MuiDataGrid-cell--textRight"
    data-colindex="6"
    data-field="places"
    role="cell"
    style="min-width: 80px; max-width: 80px; min-height: 36px; max-height: 36px;"
    tabindex="-1"
  >
    <div
      class="MuiDataGrid-cellContent"
    >
      0 / 10
    </div>
  </div>
  <div
    aria-colindex="8"
    aria-colspan="1"
    class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
    data-colindex="7"
    data-field="state"
    role="cell"
    style="min-width: 0; max-width: 0; min-height: 36px; max-height: 36px;"
    tabindex="-1"
  >
    <div
      class="MuiDataGrid-cellContent"
    >
      Julkaistu
    </div>
  </div>
</div>
`;
